hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/LongGaugeImpl.java

```diff
@@ -104,8 +104,8 @@ private static long getMetricValue(String gaugeName, Object source, ProbeFunctio
                 double doubleResult = doubleFunction.get(source);
                 return Math.round(doubleResult);
             }
-        } catch (Exception e) {
-            metricsRegistry.logger.warning("Failed to access the probe: " + gaugeName, e);
+        } catch (Throwable t) {
```

Comment:
Here we should only log the exception and rethrow using sneakyThrow everything that isn't an exception.

Reply:
943c4940847cf28642360ae5407fbe0026e3a704